spec/u_mann_whitney_spec.rb

```diff
@@ -6,29 +6,43 @@
   end
 
   describe UMannWhitney do
-    before do
-      @v1 = Daru::Vector.new([1, 2, 3, 4, 7, 8, 9, 10, 14, 15])
-      @v2 = Daru::Vector.new([5, 6, 11, 12, 13, 16, 17, 18, 19])
-      @u = UMannWhitney.new(@v1, @v2)
-    end
+    describe "with valid values" do
+      before do
+        @v1 = Daru::Vector.new([1, 2, 3, 4, 7, 8, 9, 10, 14, 15])
+        @v2 = Daru::Vector.new([5, 6, 11, 12, 13, 16, 17, 18, 19])
+        @u = UMannWhitney.new(@v1, @v2)
+      end
 
-    it 'has same result using class or Test#u_mannwhitney' do
-      expect(UMannWhitney.u_mannwhitney(@v1, @v2).u).to eq @u.u
-    end
+      it 'has same result using class or Test#u_mannwhitney' do
+        expect(UMannWhitney.u_mannwhitney(@v1, @v2).u).to eq @u.u
+      end
 
-    it 'has correct U values' do
-      expect(@u.r1).to eq 73
-      expect(@u.r2).to eq 117
-      expect(@u.u).to eq 18
-    end
+      it 'has correct U values' do
+        expect(@u.r1).to eq 73
+        expect(@u.r2).to eq 117
+        expect(@u.u).to eq 18
+      end
 
-    it 'has correct value for z' do
-      expect(@u.z).to be_within(0.001).of(-2.205)
+      it 'has correct value for z' do
+        expect(@u.z).to be_within(0.001).of(-2.205)
+      end
+
+      it 'has correct value for z and exact probability' do
+        expect(@u.probability_z).to be_within(0.001).of(0.027)
+        expect(@u.probability_exact).to be_within(0.001).of(0.028)
+      end
     end
 
-    it 'has correct value for z and exact probability' do
-      expect(@u.probability_z).to be_within(0.001).of(0.027)
-      expect(@u.probability_exact).to be_within(0.001).of(0.028)
+    describe "with invalid values" do
```

Comment:
Not familiar with the two libraries in question (`Daru` and `UMannWhiteny`), but my question is, is passing an empty array actually an "invalid" value? i.e. is passing an empty array into `UMannWhiteny.new` supposed to error because you would _never_ pass an empty array when interacting with the library?

Or should `Daru` never be supplied with an empty dataset in its initializer _or_ should it actually return an empty array instead of `nil`?

In other words are we correcting a behavior / fixing a bug or are we adjusting it to guard against an edgecase that we have in our system? The description of this spec makes me think it's the former, but, I wanna make sure that was the actual reason we made the change.

Reply:
I don't know enough about the underlying algorithm to be totally sure, but we used to pass empty arrays to it and it at least didn't blow up.